src/utils/rest-api.ts

```diff
@@ -0,0 +1,250 @@
+            error.status = response.status;
+        }
+        throw error;
+    });
+}
+
+function prepareRequest(init?: InitRequest, token?: Token): RequestInit {
+    if (!(typeof init == 'undefined' || typeof init == 'object')) {
+        throw new TypeError(
+            `Argument 2 of backendFetch is not an object ${typeof init}`
+        );
+    }
+    const initCopy: RequestInit = { ...init };
+    initCopy.headers = new Headers(initCopy.headers || {});
+    const tokenCopy = token || getToken();
+    initCopy.headers.append('Authorization', `Bearer ${tokenCopy}`);
+    return initCopy;
+}
+
+function safeFetch(url: Url, initCopy?: InitRequest): ReturnType<typeof fetch> {
+    return fetch(url, initCopy).then((response: ReqResponse) =>
+        response.ok ? response : handleError(response)
+    );
+}
+
+export function backendFetch(
+    url: Url,
+    init?: InitRequest,
+    token?: Token
+): ReturnType<typeof safeFetch> {
+    return safeFetch(url, prepareRequest(init, token));
+}
+
+export function backendFetchText(
+    url: Url,
+    init?: InitRequest,
+    token?: Token
+): ReturnType<Body['text']> {
+    return backendFetch(url, init, token).then((safeResponse: ReqResponse) =>
+        safeResponse.text()
+    );
+}
+
+export function backendFetchJson(
+    url: Url,
+    init?: InitRequest,
+    token?: Token
+): ReturnType<Body['json']> {
+    return backendFetch(url, init, token).then((safeResponse: ReqResponse) =>
+        safeResponse.json()
+    );
+}
+
+export function fetchValidateUser(user: Record<string, any>): Promise<boolean> {
+    const sub = user?.profile?.sub;
+    if (!sub) {
+        return Promise.reject(
+            new Error(
+                `Error : Fetching access for missing user.profile.sub : ${user}`
+            )
+        );
+    }
+
+    console.info(`Fetching access for user...`);
+    const CheckAccessUrl = `${PREFIX_USER_ADMIN_SERVER_QUERIES}/v1/users/${sub}`;
+    console.debug(CheckAccessUrl);
+
+    return backendFetch(CheckAccessUrl, { method: 'head' }, user?.id_token)
+        .then((response: ReqResponse) => {
+            //if the response is ok, the responseCode will be either 200 or 204 otherwise it's a Http error and it will be caught
+            return response.status === 200;
+        })
+        .catch((error) => {
+            if (error.status === 403) {
+                return false;
+            } else {
+                throw error;
+            }
+        });
+}
+
+function fetchEnv(): Promise<EnvJson> {
+    return fetch('env.json').then((res: ReqResponse) => res.json());
+}
+
+export function fetchAuthorizationCodeFlowFeatureFlag(): Promise<boolean> {
+    console.info(`Fetching authorization code flow feature flag...`);
+    return fetchEnv()
+        .then((env: EnvJson) =>
+            fetch(`${env.appsMetadataServerUrl}/authentication.json`)
+        )
+        .then((res: ReqResponse) => res.json())
+        .then((res: Record<string, any>) => {
```

Comment:
Here you can put unknown

Reply:
You mean `Record<string, unknwon>`?
I get:
>TS2322: Type Promise<unknown> is not assignable to type Promise<boolean>
>Type unknown is not assignable to type boolean

Reply:
then
```suggestion
.then(
            (res: {
                authorizationCodeFlowFeatureFlag: boolean | undefined;
            }) => {
                console.log(
                    `Authorization code flow is ${
                        res.authorizationCodeFlowFeatureFlag
                            ? 'enabled'
                            : 'disabled'
                    }`
                );
                return res.authorizationCodeFlowFeatureFlag ?? false;
            }
        )
```